lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -676,6 +676,127 @@ class DecomposeAtenTOp : public OpRewritePattern<AtenTOp> {
+
+    SmallVector<Value> repeats;
+    if (!getListConstructElements(op.repeats(), repeats))
+      return rewriter.notifyMatchFailure(
+          op, "Unimplemented: repeats not list of Scalar");
+
+    if (rank > repeats.size()) {
+      return rewriter.notifyMatchFailure(
+          op, "repeats are not matched with self's rank");
+    }
+
+    auto insertDimSizes = [](SmallVector<Value> &dimSizes,
+                             SmallVector<int64_t> &shape,
+                             const ArrayRef<Value> &vals) {
+      dimSizes.insert(dimSizes.end(), vals.begin(), vals.end());
+      std::transform(vals.begin(), vals.end(), std::back_inserter(shape),
+                     [&](Value val) -> int64_t {
+                       int64_t cst_val;
+                       if (matchPattern(val, m_TorchConstantInt(&cst_val))) {
+                         return cst_val;
+                       } else {
+                         return ShapedType::kDynamicSize;
+                       }
+                     });
+    };
+
+    Value one = rewriter.create<Torch::ConstantIntOp>(
+        loc, rewriter.getI64IntegerAttr(1));
+
+    SmallVector<Value> newSizes0, newSizes1, newSizes2;
+    SmallVector<int64_t> newShape0, newShape1;
+    // leadingRank >= 0
+    auto leadingRank = repeats.size() - rank;
+    for (size_t i = 0; i < leadingRank; ++i) {
+      insertDimSizes(newSizes0, newShape0, ArrayRef<Value>{one});
+      insertDimSizes(newSizes1, newShape1, ArrayRef<Value>{repeats[i]});
+      newSizes2.push_back(repeats[i]);
+    }
+
+    auto selfType = self.getType().dyn_cast<BaseTensorType>();
+    auto selfShape = selfType.getSizes();
+    for (size_t i = 0; i < rank; i++) {
+      auto scale = repeats[i + leadingRank];
+      Value dimSize;
+      if (selfShape[i] == ShapedType::kDynamicSize) {
+        Value dim = rewriter.create<Torch::ConstantIntOp>(
+            loc, rewriter.getI64IntegerAttr(i));
+        dimSize = rewriter.create<AtenSizeIntOp>(loc, self, dim);
+      } else {
+        dimSize = rewriter.create<Torch::ConstantIntOp>(
+            loc, rewriter.getI64IntegerAttr(selfShape[i]));
+      }
+
+      insertDimSizes(newSizes0, newShape0, ArrayRef<Value>{one, dimSize});
+      insertDimSizes(newSizes1, newShape1, ArrayRef<Value>{scale, dimSize});
+
+      Value scaledSize = rewriter.create<AtenMulIntOp>(loc, dimSize, scale);
+      newSizes2.push_back(scaledSize);
+    }
+
+    Type dtype = self.getType().cast<ValueTensorType>().getDtype();
+    Type reshapeType0 =
+        ValueTensorType::get(context, llvm::makeArrayRef(newShape0), dtype);
+    Type reshapeType1 =
+        ValueTensorType::get(context, llvm::makeArrayRef(newShape1), dtype);
+
+    auto listType = Torch::ListType::get(Torch::IntType::get(op.getContext()));
+    Value newDims0 =
+        rewriter.create<PrimListConstructOp>(loc, listType, newSizes0);
+    Value newDims1 =
+        rewriter.create<PrimListConstructOp>(loc, listType, newSizes1);
+    Value newDims2 =
```

Comment:
nit: I would call these something like `unsqueezedDims`, `expandedDims`, and `reshapedDims`